src/libpgagroal/prometheus.c

```diff
@@ -136,273 +137,330 @@ pgagroal_prometheus(int client_fd)
    exit(1);
 }
 
+int
+pgagroal_init_prometheus(size_t* p_size, void** p_shmem)
+{
+   struct configuration* config;
+
+   config = (struct configuration*) shmem;
+
```

Comment:
Create local variables for the size and the memory segment.

Set `p_size` and `p_shmem` to `0` and `NULL` in the beginning of the function, and only assign them if everything is successful